src/compiler/checker.ts

```diff
@@ -12355,6 +12427,7 @@ namespace ts {
          * @param errorNode The suggested node upon which all errors will be reported, if defined. This may or may not be the actual node used.
          * @param headMessage If the error chain should be prepended by a head message, then headMessage will be used.
          * @param containingMessageChain A chain of errors to prepend any new errors found.
+         * @param errorOutputContainer Return the diagnostic. Do not log if 'skipLogging' is truthy.

```

Comment:
`skipLogging` is new, and prevents immediate logging of the error in checkTypeRelatedTo.